### ISSUE TITLE
ParquetAvroDynamicTest is flaky

### ISSUE THREAD

**Issue by clairemcginty** [spotify/scio#5142]
Attempt 1 fails: https://github.com/spotify/scio/actions/runs/7449100380/attempts/1
Re-run succeeds: https://github.com/spotify/scio/actions/runs/7449100380/attempts/2

Error is a NPE:

```
[info] ParquetAvroDynamicTest:
[info] - should support Parquet Avro files *** FAILED *** (3 seconds, 276 milliseconds)
[info]   java.lang.NullPointerException:
[info]   at java.base/java.io.ObjectOutputStream.write(ObjectOutputStream.java:689)
[info]   at org.apache.beam.sdk.io.hadoop.SerializableConfiguration.writeExternal(SerializableConfiguration.java:72)
[info]   at java.base/java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1450)
[info]   at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1421)
[info]   at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1169)
[info]   at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1543)
[info]   at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1500)
[info]   at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1423)
[info]   at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1169)
[info]   at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:345)
```

**Comment by clairemcginty** [comment]
same issue as #5099 -- the underlying issue is Beam-related and should be fixed in next release.

**Comment by RustedBones** [comment]
Fix by #5133
